### DISCUSSION TITLE
When using body to package multiple Reducers, the type check will not work.

### DISCUSSION THREAD

**Discussion by dazy1030** [pointfreeco/swift-composable-architecture#2417] [Q&A] [answered]
hi,

I recently learned how to use bi-directional binding with BindingState, and I used body in place of the reduce method, but the type checking (completion) does not work properly. This happens even with a very simple Reducer as shown below.

```swift
struct Toggle: Reducer {
    struct State: Equatable {
        @BindingState var isToggleOn: Bool = false
    }
    
    enum Action: BindableAction, Equatable {
        case binding(BindingAction<State>)
    }
    
    var body: some Reducer<State, Action> {
        Reduce { state, aciton in
            switch aciton {
            case .binding:
                return .none
            }
        }
        BindingReducer()
    }
}
```

For example, let's add an action called `buttonPressed` to Action and add in body that it returns `Effect<Toggle.Action>.none`. At this point, I want to write something like `return .none`, but Xcode does not show the prediction. Is there a better way to do this?

**Comment by MohamedGhebaji** [comment, marked as answer]
Hi,
To fixe this you can add the generic State and Action to the reduce, like this Reduce<Sate, Action> { and autocompletion will work as expected.

**Reply by mbrandonw** [reply]
Yeah as @MohamedGhebaji mentioned, specifying the types will fix it. Also this compiler limitation is fixed in Swift 5.9 so once Xcode 15 is out this will be less of a concern.

**Reply by dazy1030** [reply]
Thank you @MohamedGhebaji! Indeed, autocomplete has come alive ☺️.
@mbrandonw, Swift is getting even smarter. I'm looking forward to Xcode15 more and more!
